cg/meta/archive/archive.py

```diff
@@ -40,33 +35,28 @@ def filter_files_on_archive_location(
     ]
 
 
+ARCHIVE_HANDLERS: Dict[str, Type[ArchiveHandler]] = {
```

Comment:
We could encapsulate/abstract away this data structure into a separate class using the registry/factory pattern. Something like 

```python
from typing import Type, Dict

class ArchiveHandlerManager:
    def __init__(self):
        self._handlers: Dict[str, Type[ArchiveHandler]] = {
            ArchiveLocations.KAROLINSKA_BUCKET: DDNDataFlowClient
        }
    # The location could be typed as an enum
    def get_archive_handler(self, location: str) -> Type[ArchiveHandler]:
        return self._handlers.get(location)

    # If you anticipate adding more handlers in the future
    def register_handler(self, location: str, handler: Type[ArchiveHandler]):
        self._handlers[location] = handler

    # ... any other necessary methods related to the handlers
```
Probably not in scope for this PR and just an idea. Feel free to disregard

Reply:
Yeah, might be an idea for the future. I will think about it!